src/toolkits/style_transfer/style_transfer_data_iterator.hpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef __TOOLKITS_STYLE_TRANSFER_H_
+#define __TOOLKITS_STYLE_TRANSFER_H_
+
+#include <functional>
+#include <map>
+#include <memory>
+
+#include <core/data/sframe/gl_sframe.hpp>
+#include <core/logging/table_printer/table_printer.hpp>
+#include <ml/neural_net/compute_context.hpp>
+#include <ml/neural_net/model_backend.hpp>
+#include <ml/neural_net/model_spec.hpp>
+#include <model_server/lib/extensions/ml_model.hpp>
+#include <toolkits/coreml_export/mlmodel_wrapper.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+struct st_image {
+  image_type style;
+  image_type content;
+  size_t index;
+};
+
+class data_iterator {
+ public:
+  struct parameters {
+    /** The Style SFrame to traverse */
+    gl_sframe style;
+
+    /** The Content SFrame to traverse */
+    gl_sframe content;
+
+    /**
+     * The name of the column containing the style images.
+     */
+    std::string style_column_name;
+
+    /**
+     * The name of the column containing the content images.
+     */
+    std::string content_column_name;
```

Comment:
Any reason not just to pass two `gl_sarray` values, one for the styles and one for the contents?

Reply:
It's the way we do it in `Python` but that's a terrible reason. I'll refactor!